### PR TITLE
Add --fixup to automatic style fix commit

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Add --fixup to automatic style fix commit - would be convenient for cleanup rebase.

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 6160a61e482715869c2c1d39c2071f61e84b2cd7 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/54774/6160a61e482715869c2c1d39c2071f61e84b2cd7/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

**Comment by Felixoid** [comment]
I don't like this idea. 1% of people does 'rebase', so this fixup will remain a garbage in the repository.

So, then we should add check that there aren't fixup comida in the PR. And it will require manual actions from contributors, and the whole idea of *automatic* fixes went to the deep south.

File: tests/ci/style_check.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def commit_push_staged(pr_info: PRInfo) -> None:
     if not git_staged:
         return
     remote_url = pr_info.event["pull_request"]["base"]["repo"]["ssh_url"]
-    git_runner(f"{GIT_PREFIX} commit -m 'Automatic style fix'")
+    git_runner(f"{GIT_PREFIX} commit -m 'Automatic style fix' --fixup HEAD")
```
Comment (Felixoid) [inline tests/ci/style_check.py:129]: I never seen the fixup argument being used together with commit message. Does it work like that? What's in the commit message then?

Comment (yakov-olkhovskiy) [reply tests/ci/style_check.py:129]: it creates commit with next content
```
fixup! <comment of referenced commit>
<comment for this commit>
```
later, on `git rebase -i <commit>` all fixup commits will be tagget as fixup

Comment (Felixoid) [reply tests/ci/style_check.py:129]: Ok, never tried it..

But still, we can't spoil the user's branch. And we shouldn't force him to do `rebase`.

It's a good intention but will meet strong resistance.

**Comment by Felixoid** [comment]
I think, having such PRs with suggestions and discussions is better than discussing it in slack. It leaves witnesses

**Pull request closed without merging**